core-java-modules/core-java-compiler/src/test/java/com/baeldung/compilerApi/JavaCompilerTest.java

```diff
@@ -0,0 +1,144 @@
+package com.baeldung.compilerApi;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemErr;
+import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class JavaCompilerTest {
+
+    @TempDir
+    static Path tempDir;
+
+    private JavaCompilerUtils compilerUtil;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        // Create a specific output directory for compiled classes
+        Path outputDir = tempDir.resolve("classes");
+        Files.createDirectories(outputDir);
+
+        // Initialize the compiler util with the output directory
+        compilerUtil = new JavaCompilerUtils(outputDir);
+    }
+
+    @Test
+    void given_simpleHelloWorldClass_when_compiledFromString_then_compilationSucceeds() {
```

Comment:
This is not the right syntax, see the suggestion for the syntax we require
```suggestion
    void givenSimpleHelloWorldClass_whenCompiledFromString_thenCompilationSucceeds() {
```